guides/v2.1/cloud/cdn/fastly-vcl-wordpress.md

```diff
@@ -11,62 +11,127 @@ functional_areas:
+-  Use the Fastly API to add the following configuration settings to the Fastly service configuration: 
 
-## Create wordpress.json {#vcl}
+   -  Add the WordPress host to the Fastly backend configuration, for example `customer.example.com`.
 
-For this example, you may only want to run it against the Production server. You can also add it to Staging for testing.
+   -  Attach the following request condition to the Wordpress backend. Incoming requests that meet this condition redirect to the WordPress backend.
 
-Create an `wordpress.json` file with the following JSON content:
+     ```json
+      req.http.X-WP == “1”
+     ```
+	 
+     See the [Fastly API reference](https://docs.fastly.com/api/config#) for details on configuring the backend and request condition.
+
+## Create an Edge Dictionary of WordPress paths {#edge-dictionary}
+
+Edge Dictionaries create key-value pairs accessible to VCL functions during VCL snippet processing. In this example, you create an edge dictionary that provides the list of URL paths that you want to redirect from your store to the WordPress backend. 
+
+1.  Log in to the Admin UI for your {{ site.data.var.ece }} project environment.
+
+1.  Navigate to **Stores** > **Configuration** > **Advanced** > **System** > **Fastly Configuration**.
+
+1.  Expand the **Edge dictionaries** section.
+
+1.  Create the Dictionary container:
+
+    - Click **Add container**.
+
+    -  On the *Container* page, enter a **Dictionary name**. For this example, use the name `wordpress_urls`.
+
+    -  Select **Activate after the change** to enable the dictionary after you create it.
+
+    -  Click **Upload** to attach the dictionary to your Fastly service configuration.
+
+1.  Add the list of URLs for redirection to the the `wordpress_urls` dictionary:
+
+    -  Click the Settings icon for the `wordpress_urls` dictonary.
+
+       ![Configure Edge Dictionary]
+
+    -  Add and save key-value pairs in the new dictionary. For this example, each **Key** is a URL path that you want to redirect to the WordPress backend, and the **Value** is 1. 
+       
+	   ![Add Edge Dictionary Items]
+	 
+    -  Click **Cancel** to return to the system configuration page.
+
+
+## Create a VCL snippet for the WordPress redirect {#vcl}
+
+The following custom VCL snippet code (JSON format) evaluates incoming requests to determine whether they require redirection to the WordPress backend you added to the Fastly service configuration.
 
 ```json
 {
-  "name": "wordpress",
+  "name": "wordpress_redirect",
   "dynamic": "0",
   "type": "recv",
   "priority": "5",
-  "content": "if ( req.url.path ~ \"^\\/?([^:\/\\s]+).*$\" ) { if ( table.lookup(wordpress_urls, re.group.1, \"NOTFOUND\") != \"NOTFOUND\" ) { set req.http.X-WP = \"1\"; } }"
+  "content": "if ( req.url.path ~ \"^\\/?([^\/?]+)") { if ( table.lookup(wordpress_urls, re.group.1, \"NOTFOUND\") != \"NOTFOUND\" ) { set req.http.X-WP = \"1\"; } }"
```

Comment:
This is invalid JSON. I realize I collaborated with you on this and I apologize if this is my mistake.

To do what the description says, unescaped value of "content" should be:
`if ( req.url.path ~ "^\/?([^\/?]+)") { if ( table.lookup(wordpress_urls, re.group.1, "NOTFOUND") != "NOTFOUND" ) { set req.http.X-WP = "1"; } }`
This can be verified here: https://regexr.com/47ee3

The JSON-escaped version of this is:
`if ( req.url.path ~ \"^\\/?([^\\/?]+)\") { if ( table.lookup(wordpress_urls, re.group.1, \"NOTFOUND\") != \"NOTFOUND\" ) { set req.http.X-WP = \"1\"; } }`
Note the extra backslash behind every quote and backslash. 

Putting this together with the key and quotes makes the correct version of line 83:
`  "content": "if ( req.url.path ~ \"^\\/?([^\\/?]+)\") { if ( table.lookup(wordpress_urls, re.group.1, \"NOTFOUND\") != \"NOTFOUND\" ) { set req.http.X-WP = \"1\"; } }"`